### ISSUE TITLE
Low Accuracy When Classifying Van Gogh-Style Images Generated by Stable Diffusion 1.4 or 1.5

### ISSUE THREAD

**Issue by ryliu68** [OPTML-Group/UnlearnCanvas#20]
Hi, I'm encountering **unexpectedly low accuracy (~10%)** when using a pretrained style classifier to recognize **Van Gogh-style images** generated by **Stable Diffusion 1.4 and 1.5**.

---

## 🖼️ Image Generation

I generated a set of images using prompts that explicitly request *"Van Gogh style"*, for example:

- "A portrait in Van Gogh style"
- "A Van Gogh landscape painting"

---

## 📊 Classification Results

I evaluated 1000 such images using my classifier, but the recognition performance is very low:

- **Accuracy**: `10.9%`

- **Predicted label distribution**:

```python
Counter({
    14: 257, 38: 206, 25: 203, 45: 109, 46: 45, 17: 38, 13: 33, 15: 27,
    23: 19, 41: 13, 7: 9, 2: 8, 3: 7, 35: 4, 19: 4, 28: 3, 18: 3, 43: 2,
    42: 2, 6: 2, 34: 1, 31: 1, 39: 1, 49: 1, 21: 1, 48: 1
})
```

---

## ❓ My Questions

- Is this **expected behavior** for images generated by SD 1.4 / 1.5?
- Could there be **issues in my image processing pipeline**?
- Has anyone experienced similar behavior?

---

## 🧪 Classifier Code

```python
from UnlearnCanvas_const import theme_available, class_available
from PIL import Image
import argparse
import timm
from torchvision import transforms
import torch
import glob

device = "cuda" if torch.cuda.is_available() else "cpu"

if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument("--ckpt", type=str, default="ckpt/canvas/cls_model/style50.pth")
    args = parser.parse_args()

    classifer = timm.create_model("vit_large_patch16_224.augreg_in21k", pretrained=True).to(device)
    num_classes = len(theme_available) if args.task == "style" else len(class_available)
    classifer.head = torch.nn.Linear(1024, num_classes).to(device)
    classifer.load_state_dict(torch.load(args.ckpt, map_location=device)["model_state_dict"])
    classifer.eval()

    image_transform = transforms.Compose([
        transforms.Resize((224, 224)),
        transforms.ToTensor(),
        transforms.Normalize([0.5], [0.5]),
    ])

    theme_label = theme_available.index("Van_Gogh")
    filenames = glob.glob("data/gen_imgs/sd15/*[.png|.jpg|.jpeg|.JPEG]")

    acc = 0
    total_imgs = len(filenames)
    for filename in filenames:
        image = Image.open(filename)
        target_image = image_transform(image).unsqueeze(0).to(device)
        with torch.no_grad():
            res = classifer(target_image)
            res_softmax = torch.nn.functional.softmax(res, dim=1)
            pred_label = torch.argmax(res)
            if pred_label.item() == theme_label:
                acc += 1

    accuracy = round(100 * acc / total_imgs, 2)
    print(accuracy)
```

Any suggestions on what might be going wrong? Or is SD 1.4 or 1.5 simply not good at capturing recognizable Van Gogh styles for classification?

Thanks in advance!
